### ISSUE TITLE
"Execute SQL" option crash using AZURE MS SQL Server connection

### ISSUE THREAD

**Issue by tschlarman** [qgis/QGIS#49340]
## User Feedback

1) R-click on empty Azure SQL table with shape enabled column (geography column)
2) Chose "Execute SQL"

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Thread 0x00004204 (most recent call first):
<no Python frame>
```

**Stack Trace**
No stack trace is available.

**QGIS Info**
QGIS Version: 3.26.0-Buenos Aires
QGIS code revision: 0aece2818e
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.5.0
Running against GDAL: 3.5.0

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.22000

### Steps to reproduce the issue

1) R-click on empty Azure SQL table with shape enabled column (geography column)
2) Chose "Execute SQL"

### Versions

QGIS Version: 3.26.0-Buenos Aires
QGIS code revision: 0aece2818e
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.5.0
Running against GDAL: 3.5.0

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by gioman** [comment]
@tschlarman what is te backend? PostGIS? MS SQL Server?

**Comment by tschlarman** [comment]
It was an Azure SQL table with a SQL Server backend.

-----------------------
Tom Schlarman
***@***.***

________________________________

**Comment by gioman** [comment]
> It was an Azure SQL table with a SQL Server backend.

@tschlarman is the error arising only when you use SQL Server on Azure, or is the same with a local installation (for example)?

**Comment by tschlarman** [comment]
local SQL Server seems fine

**Comment by gioman** [comment]
Something special about MS SQL Server on Azure?

**Comment by tschlarman** [comment]
It also happens with a non-empty table and with v3.26.1

I'm sure there are some differences between the two but nothing in normal TSQL querying. What is 'Execute SQL...' doing?

Same error in v3.26.1:

Python Stack Trace
Windows fatal exception: access violation

Thread 0x00002c94 (most recent call first):

Stack Trace
No stack trace is available.

QGIS Info
QGIS Version: 3.26.1-Buenos Aires
QGIS code revision: b609df9ed4d
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.5.1
Running against GDAL: 3.5.1

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.22000

**Comment by gioman** [comment]
> What is 'Execute SQL...' doing?

i.e. apply a select/filter on the backend side before loading the table in QGIS.

**Comment by tschlarman** [comment]
I'm sure I'm missing something, but it seems like it might doing a bit more than needed at dialog startup if all it shows is a dialog with a couple of text boxes and buttons. I wouldn't expect it hit the DB until the user hits the 'Execute' button.

![image](https://user-images.githubusercontent.com/10763706/183716036-e5fbff96-c7cf-4713-9124-0d0d4c2ca726.png)

(picture from a local DB where Execute SQL dialog shows up)

**Comment by tschlarman** [comment]
This is fixed in 3.26.2. It does however uses a MySQL query and not a SQL Server Query:

SELECT * FROM "dbo"."EventActive" LIMIT 10

For SQL Server the query should be:

SELECT top 10 * FROM "dbo"."EventActive"

I will open a new issue

**Comment by tschlarman** [comment]
Is closed in 3.26.2
